backend/modules/parsers/videoparser.py

```diff
@@ -0,0 +1,162 @@
+                "prompt": "Descibe the given information present in the image - textual/charts/graphs/tables in detail."
+                "total_frames": 15 // total images to be process from video, for testing puposes, if not provided all the images will be processed
+            }
+        }
+    }
+    """
+
+    supported_file_extensions = [
+        ".mp4",
+        ".avi",
+        ".mov",
+        ".flv",
+        ".wmv",
+        ".webm",
+        ".mkv",
+        ".mpg",
+        ".mpeg",
+        ".ogv",
+        ".3gp",
+    ]
+
+    def __init__(
+        self,
+        vlm_model_configuration: ModelConfig,
+        audio_model_configuration: ModelConfig,
+        max_chunk_size: int = 2000,
+        prompt: str = "",
+        total_frames: int = -1,
+        **kwargs,
+    ):
+        """
+        Initializes the VideoParser object.
+        """
+
+        self.vlm_model_configuration = ModelConfig.model_validate(
+            vlm_model_configuration
+        )
+        self.audio_model_configuration = ModelConfig.model_validate(
+            audio_model_configuration
+        )
+        self.max_chunk_size = max_chunk_size
+        self.prompt = prompt
+        self.total_frames = (
+            total_frames  # total images to be process from video, for testing puposes
+        )
+
+        super().__init__(**kwargs)
+
+    def _get_images_from_video(self, video_filepath: str) -> str:
+        """
+        Extract images from video frames,
+        We extract 1 frame every 5 seconds of the video
+        Save all the images in temporary directory
+        """
+        _, file_name = os.path.split(video_filepath)
+        file_name, _ = os.path.splitext(file_name)
+        clip = VideoFileClip(video_filepath)
+        temp_images_dir = tempfile.mkdtemp()
+        clip.write_images_sequence(
+            os.path.join(temp_images_dir, file_name + "_frame%04d.png"),
+            fps=0.2,  # configure this for controlling frame rate.
+        )
+        return temp_images_dir
+
+    def _get_audio_from_video(self, video_filepath: str) -> str:
+        """
+        Extract audio from video
+        """
+        _, file_name = os.path.split(video_filepath)
+        # Save the audio in a temporary directory
+        temp_audio_dir = tempfile.mkdtemp()
+        clip = VideoFileClip(video_filepath)
+        clip.audio.write_audiofile(os.path.join(temp_audio_dir, file_name + ".wav"))
+        return temp_audio_dir
+
+    async def get_chunks(
+        self, filepath: str, metadata: Dict[Any, Any] | None, **kwargs
+    ) -> List[Document]:
+        """
+        Get the chunks of the video file.
+        """
+        try:
+            # Extract images from video
+            temp_images_dir = self._get_images_from_video(filepath)
+
+            # Init multimodal parser for extracting text from images
+            multimodal_parser = MultiModalParser(
+                model_configuration=self.vlm_model_configuration, prompt=self.prompt
```

Comment:
max_chunk_size is not being propagated here?

Reply:
Currently we are not doing any chunking in `MultiModalParser`. 
We can later on gather all text and pass it to `UnstructuredIoParser`